TFT/src/User/API/Gcode/mygcodefs.c

```diff
@@ -112,14 +112,14 @@ bool scanPrintFilesGcodeFs(void)
         else
           Pstr_tmp++;
 
-        infoFile.longFile[infoFile.fileCount] = malloc(strlen(Pstr_tmp) + 2);  // plus one extra byte for filename extension check
+        infoFile.longFile[infoFile.fileCount] = malloc(strlen(Pstr_tmp) + 2);  // "+ 2": space for terminating null character and the flag for filename extension check
         if (infoFile.longFile[infoFile.fileCount] == NULL)
         {
           clearRequestCommandInfo();
           break;
         }
-        strcpy(infoFile.longFile[infoFile.fileCount], Pstr_tmp);
-        infoFile.longFile[infoFile.fileCount][strlen(Pstr_tmp) + 1] = 0;  // set to 0 the extra byte for filename extension check
+        // copy file name and set the flag for filename extension check
+        strncpy(infoFile.longFile[infoFile.fileCount], Pstr_tmp, strlen(Pstr_tmp) + 2);  // "+2":  terminating null character and the flag for filename extension check
```

Comment:
// "+ 2": space for terminating....     instead of  // "+2":  terminating....

Reply:
Done